packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -826,6 +826,22 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
      */
     private delayConnectClientId?: string;
 
+    private runWithoutOpsCalls = 0;
+
+    /**
+     * Invokes the given callback and expects that no ops are submitted
+     * until execution finishes. If an op is submitted, an error will be raised.
+     *
+     * Can be disabled by feature gate `Fluid.ContainerRuntime.DisableOpReentryCheck`
+     *
+     * @param callback - the callback to be invoked
+     */
+    public runWithoutOps(callback: () => void): void {
+        this.runWithoutOpsCalls++;
+        callback();
```

Comment:
It does not handle exceptions well - this.runWithoutOpsCalls-- should be in finally clause of try/catch